salt/modules/zabbix.py

```diff
@@ -1471,8 +1471,10 @@ def host_inventory_set(hostid, **connection_args):
 
             if connection_args.get("clear_old"):
                 clear_old = True
-
             connection_args.pop("clear_old", None)
+
+            inventory_mode = connection_args.pop("inventory_mode", "0")
```

Comment:
**question**: does inventory_mode need to be removed from connection_args?

Reply:
The `connection_args`, despite its name, carriers all kind of information, not only connection arguments like server, user and password. The inventory items are inside `connection_args` and needs to be moved to the element `inventory` before sent to Zabbix API, the `inventory_mode` is also inside `connection_args` but it's not an inventory item, it needs to be at same level of `inventory`, not inside it.

So, the parameters needs to be moved around to transform a flat set of parameters in the hierarchy expected by `host.update`. The `inventory_mode` pop is part of that parameters dance.